examples/OcclusionSaliency.ipynb

```diff
@@ -0,0 +1,211 @@
+      "Saving to: ‘imagenet_classes.txt.4’\n",
+      "\n",
+      "imagenet_classes.tx 100%[===================>]  10.23K  --.-KB/s    in 0s      \n",
+      "\n",
+      "2021-05-20 16:56:54 (48.0 MB/s) - ‘imagenet_classes.txt.4’ saved [10472/10472]\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "\"\"\"\n",
+    "This notebook contains a working example show usage of the new\n",
+    "image classifier based saliency map API for choosing to use \n",
+    "a occlusion based saliency map algorithm.\n",
+    "\n",
+    "Reference for implementation: Zeiler MD, Fergus R. Visualizing \n",
+    "and understanding convolutional networks. InEuropean conference\n",
+    "on computer vision 2014 Sep 6 (pp. 818-833). Springer, Cham.\n",
+    "\n",
+    "\"\"\"\n",
+    "from xaitk_saliency.interfaces.perturb_image import PerturbImage\n",
+    "from xaitk_saliency.interfaces.vis_sal_classifier import ImageClassifierSaliencyMapGenerator\n",
+    "from smqtk_core.configuration import from_config_dict\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "from PIL import Image\n",
+    "from torchvision import transforms\n",
+    "from torch.autograd import Variable\n",
+    "import torchvision.models as models\n",
+    "from sklearn.linear_model import LogisticRegression\n",
+    "import torch\n",
+    "from sklearn.preprocessing import minmax_scale\n",
+    "import numpy as np\n",
+    "from operator import itemgetter\n",
+    "\n",
+    "# Test image to be explained\n",
+    "test_image = 'catdog.png'\n",
+    "# Imagenet class for which saliency maps need to be computed \n",
+    "sal_classes = ['bull mastiff', 'Egyptian cat']\n",
+    "\n",
+    "# Declaring xaitk implementation instantiations \n",
+    "\n",
+    "sliding_window_cfg = {\n",
+    "    \"type\": \"xaitk_saliency.impls.perturb_image.occlusion.OcclusionBasedPerturb\",\n",
+    "    \"xaitk_saliency.impls.perturb_image.occlusion.OcclusionBasedPerturb\": {\n",
+    "    \"window_size\": 25,\n",
+    "    \"stride\": 5,\n",
+    "    },\n",
+    "  }\n",
+    "occ_cfg = {\n",
+    "    \"type\": \"xaitk_saliency.impls.vis_sal_classifier.occlusion_scoring.OcclusionScoring\",\n",
+    "    \"xaitk_saliency.impls.vis_sal_classifier.occlusion_scoring.OcclusionScoring\": {\n",
+    "    },\n",
+    "  }\n",
+    "\n",
+    "# Creating xaitk objects for sliding window and classifier based Fsal approach. \n",
+    "slid_algo = from_config_dict(sliding_window_cfg, PerturbImage.get_impls())\n",
+    "occ_alg = from_config_dict(occ_cfg, ImageClassifierSaliencyMapGenerator.get_impls())\n",
+    "\n",
+    "# Creating an example deep feature extractor to describe images\n",
+    "model = models.resnet18(pretrained=True)\n",
+    "model = model.eval()\n",
+    "model = model.cuda()\n",
+    "\n",
+    "# Download ImageNet labels and extract id for required classes\n",
+    "!wget https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt\n",
```

Comment:
I missed this before, but we should add this file path/name to the `.gitignore` file (or alternatively create one for this directory) to prevent this from getting mistakenly committed to this repo.

Reply:
Add a .gitignore for this specific directory.

Reply:
The pattern you added is a bit overkill, but it is fine for now.